src/JPEGView/HEIFWrapper.cpp

```diff
@@ -69,5 +71,19 @@ void * HeifReader::ReadImage(int &width,
 	}
 	ICCProfileTransform::DeleteTransform(transform);
 
+	std::vector<heif_item_id> exif_blocks = handle.get_list_of_metadata_block_IDs("Exif");
+
+	if (!exif_blocks.empty()) {
+		std::vector<uint8_t> exif = handle.get_metadata(exif_blocks[0]);
+		if (exif.size() > 8 && exif.size() < 65538) {
```

Comment:
where's the magic 65538 number come from?  I see it in other EXIF chunks of the code, where size+some number is 65538.  It's an odd number, not a power of 2?

Reply:
These libraries all have their own ideas about where to start the Exif data from. JPEG Exif blocks are in the format `FF E1 SS SS 45 78 69 66 00 00 [data]`. The `SS SS` is a big-endian unsigned short representing the size of everything after `FF E1`. 

libjxl gives `00 00 00 00 [data]`, libheif gives `00 00 00 00 45 78 69 66 00 00 [data]`, libavif, libwebp and libpng give `[data]`, so they have different limits for size. If you want I can change it to 65536 + an offset and add notes explaining why.

Reply:
I see, now it makes sense.  Regarding the documentation, I'll just refer to this comment.  This is good investigative information.